Packs/TOPdesk/Integrations/TOPdesk/TOPdesk_test.py

```diff
@@ -0,0 +1,816 @@
+    (list_operators_command,
+     {'query': 'id==1st-operator-id'},
+     ('https://test.com/api/v1/operators?query=id==1st-operator-id', {})),
+    (list_operators_command,
+     {'page_size': 2, 'start': 2, 'query': 'id==1st-operator-id'},
+     ('https://test.com/api/v1/operators?start=2&page_size=2&query=id==1st-operator-id', {})),
+    (list_operators_command,
+     {'page_size': 2, 'query': 'id==1st-operator-id'},
+     ('https://test.com/api/v1/operators?page_size=2&query=id==1st-operator-id', {})),
+    (list_persons_command,
+     {'page_size': 2},
+     ('https://test.com/api/v1/persons?page_size=2', {})),
+    (list_persons_command,
+     {'start': 2},
+     ('https://test.com/api/v1/persons?start=2', {})),
+    (list_persons_command,
+     {'query': 'id==1st-person-id'},
+     ('https://test.com/api/v1/persons?query=id==1st-person-id', {})),
+    (list_persons_command,
+     {'page_size': 2, 'start': 2, 'query': 'id==1st-person-id'},
+     ('https://test.com/api/v1/persons?start=2&page_size=2&query=id==1st-person-id', {})),
+    (list_persons_command,
+     {'page_size': 2, 'query': 'id==1st-person-id'},
+     ('https://test.com/api/v1/persons?page_size=2&query=id==1st-person-id', {})),
+    (get_incidents_list_command,
+     {'page_size': 2},
+     ('https://test.com/api/v1/incidents?page_size=2', {})),
+    (get_incidents_list_command,
+     {'start': 2},
+     ('https://test.com/api/v1/incidents?start=2', {})),
+    (get_incidents_list_command,
+     {'query': 'id==1st-incident-id'},
+     ('https://test.com/api/v1/incidents?query=id==1st-incident-id', {})),
+    (get_incidents_list_command,
+     {'page_size': 2, 'start': 2, 'query': 'id==1st-incident-id'},
+     ('https://test.com/api/v1/incidents?start=2&page_size=2&query=id==1st-incident-id', {})),
+    (get_incidents_list_command,
+     {'page_size': 2, 'query': 'id==1st-incident-id'},
+     ('https://test.com/api/v1/incidents?page_size=2&query=id==1st-incident-id', {})),
+    (get_incidents_list_command,
+     {'page_size': 2, 'category': 'some_category'},
+     ('https://test.com/api/v1/incidents?page_size=2&query=category==some_category', {}))
+
+])
+def test_list_command_with_args(requests_mock,
+                                command,
+                                command_args,
+                                command_api_request):
+    """Unit test
+    Given
+        - command args
+    When
+        - running the command
+    Then
+        - validate the correct request was called.
+        - validate the request body is as expected
+    """
+    client = Client(
+        base_url='https://test.com/api/v1',
+        verify=False,
+        auth=('some_username', 'some_password'),
+        new_query=True
+    )
+    requests_mock.get(
+        command_api_request[0], json=[{}])
+    command(client, command_args)
+
+    assert requests_mock.called
+    assert requests_mock.last_request.json() == command_api_request[1]
```

Comment:
not sure i understand why this is needed, these are all get requests with no body

Reply:
This is good for testing the assembly of the inline parameters of the request. Since the logic of this assembly is done within the integration code I felt the need to check it, especially since the query parameter is a bit tricky at times. If you feel this is redundant, I'll delete it.

Reply:
you can keep it if you'd like, just wondered